### ISSUE TITLE
Display broken of all day events created on the last Sunday of March and October (switching DST)

### ISSUE THREAD

**Issue by GoogleCodeExporter** [roundcubevnz/myroundcube#231]
```
In version 1.1 (0.5.3 bundle) the Calendar plugin worked quite well with 
recurring all day events created on the last Sunday of March or the last Sunday 
in October (switching DST). Since version 1.2 it's broken. Checked on site 
http://mail4us.net with user 'jackcal' (26 July, 21:00-22:30 CEST). The 
testdata from the last time(s) were still there and haven't been changed. 

What I see: Recurring all day events created on the last Sunday of March last 
25 hrs. Recurring all day events created on the last Sunday of October last 23 
hrs. Sometimes the “hover” end time is OK but the “double click” end 
time is not. Data in database is OK so it's a display issue.


What I've tested:
* Last Sunday in March (example 27 March 2011)
Test 1B (yearly): 27 March 2011
hover over event: start 0:00:00, end 0:59:00 (lasts 25 hrs instead of 24 hrs)
double click on event: same issue

Test 1B (yearly): 27 March 2012 (and all following years)
hover over event: start 0:00:00, end 23:59:00 (this is OK)
double click on event: start 0:00:00, end 0:59:00 (lasts 25 hrs instead of 24 
hrs)

Test 2B (monthly): 27 March 2011
hover over event: start 0:00:00, end 0:59:00 (lasts 25 hrs instead of 24 hrs)
double click on event: same issue

Test 2B (monthly): 27 April 2011
hover over event: start 0:00:00, end 23:59:00 (this is OK)
double click on event: start 0:00:00, end 0:59:00 (lasts 25 hrs instead of 24 
hrs)

Test 3B (weekly): 27 March 2011
hover over event: start 0:00:00, end 0:59:00 (lasts 25 hrs instead of 24 hrs)
double click on event: same issue

Test 3B (weekly): 27 March 2011
hover over event: start 0:00:00, end 0:59:00 (lasts 25 hrs instead of 24 hrs)
double click on event: same issue

Test 4B (daily): 27 March 2011
hover over event: start 0:00:00, end 0:59:00 (lasts 25 hrs instead of 24 hrs)
double click on event: same issue

Test 4B (daily): 28 March 2011
hover over event: start 0:00:00, end 23:59:00 (this is OK)

Test 4B (daily): 30 October 2011
hover over event: start 0:00:00, end 22:59:00 (lasts 23 hrs instead of 24 hrs)


* Last Sunday in October (example 30 October 2011)
Test 1A (yearly):  30 October 2011
hover over event: start 0:00:00, end 22:59:00 (lasts 23 hrs instead of 24 hrs)
double click on event: same issue

Test 2A (monthly): 30 October 2011
hover over event: start 0:00:00, end 22:59:00 (lasts 23 hrs instead of 24 hrs)
double click on event: same issue

Test 3A (weekly):  30 October 2011
hover over event: start 0:00:00, end 22:59:00 (lasts 23 hrs instead of 24 hrs)
double click on event: same issue

Test 4A (daily):  30 October 2011
hover over event: start 0:00:00, end 22:59:00 (lasts 23 hrs instead of 24 hrs)
double click on event: same issue



```

Original issue reported on code.google.com by `hijack...@gmail.com` on 26 Jul 2011 at 8:48

**Comment by GoogleCodeExporter** [comment]
```
fixed in Roundcube-0.5.3-bundle-v1.4.zip

Background:

I added faked allday events like in Thunderbird. This means an event which 
spans over more than one day is displayed in the allday section even if it has 
no allday flag. I missed that events on DST change have a duration of 86400 + 
3600 or -3600 seconds.
```

Original comment by `rol...@liebl.ath.cx` on 30 Jul 2011 at 6:18
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Thanks!

-------- Oorspronkelijk bericht --------
Van:  myroundcube@googlecode.com
Verzonden:  zaterdag 30 juli 2011 8:18:55
Aan:  hijack572@gmail.com
Onderwerp:  Re: Issue 231 in myroundcube: Display broken of all day
events created on the last Sunday of March and October (switching DST)
```

Original comment by `hijack...@gmail.com` on 30 Jul 2011 at 8:05
